### PR TITLE
Autodetect ESM supports during asinit

### PR DESCRIPTION
**Pull request by MaxGraey**
- [x] I've read the contributing guidelines

### CODE REVIEW

File: bin/asinit [hunk 1]
```diff
@@ -377,9 +377,14 @@ function ensurePackageJson() {
 function ensureIndexJs() {
   console.log("- Making sure that 'index.js' exists...");
   if (!fs.existsSync(indexFile)) {
+    // since node.js v13.2.0 or v12.17.0 we can use ESM without flags
+    const ver = process.versions.node.split('.');
+    const maj = parseInt(ver[0]);
+    const min = parseInt(ver[1]);
+    const supportESM = (maj >= 12 && min >= 17 && maj < 13) || (maj >= 13 && min >= 2);
```
Comment (dcodeIO) [inline bin/asinit:384]: Does this cover versions like 14.0?

Comment (MaxGraey) [reply bin/asinit:384]: It should. It covers all  from`12.17` to `13.0` (exclusive) and from `13.2` to `latest`

Comment (dcodeIO) [reply bin/asinit:384]:
```
> maj = 14
14
> min = 0
0
> (maj >= 12 && min >= 17 && maj < 13) || (maj >= 13 && min >= 2)
false
```

```
> maj >= 14 || (maj == 12 && min >= 17) || (maj == 13 && min >= 2)
true
```

Comment (MaxGraey) [reply bin/asinit:384]: right! Will fix this

File: bin/asinit [hunk 2]
```diff
@@ -377,9 +377,14 @@ function ensurePackageJson() {
 function ensureIndexJs() {
   console.log("- Making sure that 'index.js' exists...");
   if (!fs.existsSync(indexFile)) {
+    // since node.js v13.2.0 or v12.17.0 we can use ESM without flags
+    const ver = process.versions.node.split('.');
+    const maj = parseInt(ver[0]);
+    const min = parseInt(ver[1]);
+    const supportESM = maj >= 14 || (maj >= 13 && min >= 2) || (maj >= 12 && min >= 17 && maj < 13);
```
Comment (dcodeIO) [inline bin/asinit:384]:
```suggestion
    const supportESM = maj >= 14 || (maj == 13 && min >= 2) || (maj == 12 && min >= 17);
```

**Pull request merged**